staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -494,11 +585,29 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 
 	if oldCRD.UID != newCRD.UID {
 		r.removeStorage_locked(oldCRD.UID)
+		r.stopStorageVersionUpdateQueue_locked(oldCRD.UID)
 	}
 
 	storageMap := r.customStorage.Load().(crdStorageMap)
 	oldInfo, found := storageMap[newCRD.UID]
 	if !found {
+		if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+			utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+			q := r.getOrCreateStorageVersionUpdateQueue_locked(newCRD.UID)
+			if len(q) == maxStorageVersionUpdateRequests {
+				// When the channel buffer is full, writing to the channel becomes blocking.
```

Comment:
when can this happen? On storage version changes should be enough, shouldn't it? Now it's on every update of the CRD?

Reply:
update happens at:

1. new storage creation
2. CRD update event: old storage doesn't exist
3. CRD update event: old storage exists, and the storage version changes
4. CRD create event: old storage doesn't exist
5. CRD create event: old storage exists, and the storage version changes

2, 3, 4, 5 exist only to make sure if there is no CR request, or if CR requests aren't hitting all servers in an HA cluster, the StorageVersion record for each server still gets updated.

we don't send update requests when:
1. CRD update event: old storage with same storage version exists
2. CRD create event: old storage with same storage version exists